File: jmlparser-jml-tools/src/main/java/com/github/jmlparser/jml2java/J2JCommand.java
package com.github.jmlparser.jml2java;

import com.beust.jcommander.Parameter;
import com.beust.jcommander.Parameters;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

/**
 * @author Alexander Weigl
 * @version 1 (09.04.23)
 */
@Parameters(commandNames = {"jml2java"},
        commandDescription = "Submit usage for a given customer and subscription, accepts one usage item")
public class J2JCommand {
    @Parameter(names = {"-o", "--output"}, description = "Output folder of the Java Files")
    private File outputFolder = new File("out");

    @Parameter(names = {"--jbmc"}, description = "JBMC mode")
    private boolean jjbmcMode = false;
    @Parameter(description = "FILES")
    private List<String> files = new ArrayList<>();
}

File: tools/cli/src/main/kotlin/io/github/jmltoolkit/cli/J2JCommand.kt
package io.github.jmltoolkit.cli

import com.github.ajalt.clikt.core.CliktCommand
import com.github.ajalt.clikt.core.Context
import com.github.ajalt.clikt.parameters.arguments.argument
import com.github.ajalt.clikt.parameters.arguments.multiple
import com.github.ajalt.clikt.parameters.options.default
import com.github.ajalt.clikt.parameters.options.option
import com.github.ajalt.clikt.parameters.types.file
import java.io.File

/**
 * @author Alexander Weigl
 * @version 1 (09.04.23)
 */
class J2JCommand :
    CliktCommand("jml2java") {
    override fun help(context: Context): String =
        "Submit usage for a given customer and subscription, accepts one usage item"

    private val outputFolder by option("-o", "--output", help = "Output folder of the Java Files")
        .file()
        .default(File("out"))

    private val jjbmcMode by option("jjbmc", "JBMC mode")
    private val files by argument("FILES").file().multiple()
    override fun run() {
        TODO("Not yet implemented")
    }
}
